packages/sil-voting-app/src/routes/dashboard/components/Proposal/VotingPanel/Buttons.tsx

```diff
@@ -1,34 +1,59 @@
 import { VoteOption } from "@iov/bns";
 import Button from "@material-ui/core/Button";
-import { Block } from "medulas-react-components";
-import React from "react";
+import { Block, Form, useForm } from "medulas-react-components";
+import React, { useState } from "react";
+import * as ReactRedux from "react-redux";
+
+import { sendSignAndPostRequest } from "../../../../../communication/signandpost";
+import { getBnsConnection } from "../../../../../logic/connection";
+import { RootState } from "../../../../../store/reducers";
 
 interface Props {
+  readonly id: number;
   readonly vote: VoteOption | undefined;
 }
 
-const Buttons = ({ vote }: Props): JSX.Element => {
-  const yesButton = vote === VoteOption.Yes ? "contained" : "outlined";
-  const noButton = vote === VoteOption.No ? "contained" : "outlined";
-  const abstainButton = vote === VoteOption.Abstain ? "contained" : "outlined";
+const Buttons = ({ id, vote }: Props): JSX.Element => {
+  const [currentVote, setCurrentVote] = useState(vote);
+  const governor = ReactRedux.useSelector((state: RootState) => state.extension.governor);
+
+  const yesButton = currentVote === VoteOption.Yes ? "contained" : "outlined";
+  const noButton = currentVote === VoteOption.No ? "contained" : "outlined";
+  const abstainButton = currentVote === VoteOption.Abstain ? "contained" : "outlined";
+
+  const voteYes = (): void => setCurrentVote(VoteOption.Yes);
+  const voteNo = (): void => setCurrentVote(VoteOption.No);
+  const voteAbstain = (): void => setCurrentVote(VoteOption.Abstain);
+
+  const submitVote = async (): Promise<void> => {
+    if (governor && currentVote) {
```

Comment:
This should throw an exception when `governor` is not set as it is a bug in the app